### PR TITLE
HW6: monolith vs microservices

### PR DESCRIPTION
**Pull request by Hukyl**
I have decided to go with modular monolith app architecture, and most of my apps already resemble this structure.

From previous PR:
- updated project layout to be resemble one in Go docs

In this PR:
- updated emailing: sending to multiple users is now done in one message using blind carbon copy (BCC).

Would extremely appreciate feedback on possible improvements!

### CODE REVIEW

File: internal/mail/backends/gomail_test.go [hunk 1]
```diff
@@ -55,65 +55,98 @@ func TestSendEmail_InvalidPort(t *testing.T) {
...
+				return
+			}
+			require.NoError(t, err)
+			messages := smtpServer.Messages()
+			assert.Len(t, messages, 1)
+		})
+	}
 }
 
 func TestVariousParameters(t *testing.T) {
```
Comment (hrvadl) [inline internal/mail/backends/gomail_test.go:97]:
```suggestion
func TestSendEmailWithVariousParameters(t *testing.T) {
```
maybe give it a more descriptive name? What exactly do you test with various parameters?

File: internal/mail/client.go [hunk 2]
```diff
@@ -6,7 +6,7 @@ import (
 )
 
 type Mailer interface {
-	SendEmail(ctx context.Context, email, subject, message string) error
+	SendEmail(ctx context.Context, emails []string, subject, message string) error
```
Comment (hrvadl) [inline internal/mail/client.go:9]: cool, it's nice that you added support for multiple receivers

File: internal/mail/backends/gomail.go [hunk 3]
```diff
@@ -2,21 +2,29 @@ package backends
...
+	_ context.Context, emails []string,
+	subject, message string,
+) error {
 	mail := gomail.NewMessage()
 	mail.SetHeader("From", gm.config.FromEmail)
-	mail.SetHeader("To", email)
+	if len(emails) == 0 {
+		return errors.New("no email recipients")
+	}
+	mail.SetHeader("To", emails[0])
```
Comment (hrvadl) [inline internal/mail/backends/gomail.go:26]: why don't you send all as BCC?

File: internal/mail/backends/gomail.go [hunk 4]
```diff
@@ -2,21 +2,29 @@ package backends
...
+	"github.com/Hukyl/genesis-kma-school-entry/internal/mail/config"
 	"github.com/go-gomail/gomail"
 )
 
 type GomailMailer struct {
 	config config.Config
 }
 
-func (gm *GomailMailer) SendEmail(_ context.Context, email, subject, message string) error {
+func (gm *GomailMailer) SendEmail(
```
Comment (hrvadl) [inline internal/mail/backends/gomail.go:17]: wait, you require a context to be passed, but don't use it? What's the point then?

**Review by hrvadl — COMMENTED** [review]
All comments are optional, except one - with context. Overall this is looking good, nice project structure and good that you've added support for multiple mail receivers

File: internal/mail/backends/gomail.go (hunk 4 above)
Comment (Hukyl) [reply internal/mail/backends/gomail.go:17]: solved in next PR to avoid merge conflicts

**Review by KushnerykPavel — APPROVED** [review]

**Pull request merged**